### ISSUE TITLE
Instructions on how to install this with docker?

### ISSUE THREAD

**Issue by DevJackSmith** [getsentry/sentry-auth-google#27]
Hi I followed the instructions on how to setup sentry from here:
https://github.com/getsentry/onpremise
And figured out that for email to work i need to add environment variables to the docker-compose.yml file. But just adding GOOGLE_CLIENT_ID and GOOGLE_CLIENT_SECRET to the same place does not work.

Do I need to download something somewhere? Or run the pip command inside a container? or how do I install it?

**Comment by DevJackSmith** [comment]
- tried manually installing inside web container
- tried adding inside sentry.config.py
- tried adding RUN pip install command and COPY sentry.config.py commands inside Dockerfile
- tried searching on google, sentry docs, asking on irc #sentry
No luck so far.

**Comment by DevJackSmith** [comment]
Ok I figured it out:
After installing sentry onpremise and it's working, to install this plugin i had to do the following:
**1.** Add the following line to requirements.txt:
`https://github.com/getsentry/sentry-auth-google/archive/52020f577f587595fea55f5d05520f1473deaad1.zip`
**2.** Add the following two lines in docker-compose.yml under services: base: environment:
`GOOGLE_CLIENT_ID: <your id from your google project>`
`GOOGLE_CLIENT_SECRET: <your secret from your google project>`
**3.** Run the following 3 commands:
`docker-compose down`
`docker-compose build` (just trying `make build` did not work)
`docker-compose up -d`

**Comment by DevJackSmith** [comment]
can we maybe update readme?

**Comment by jeeshofone** [comment]
yes, please update the readme. This worked for me too!

Thank you, @DevJackSmith!

**Comment by DevJackSmith** [comment]
in the newest version the above solution will give you a warning. Instead, I added the following to sentry.conf.py
```
SENTRY_OPTIONS['auth-google.client-id'] = 'your_own_client_id_here'
SENTRY_OPTIONS['auth-google.client-secret'] = 'your_own_client_secret_here'
```

**Comment by silvez-dh** [comment]
For me on 9.1.0 only worked when i set on config.py:

```
GOOGLE_CLIENT_ID = env('GOOGLE_CLIENT_ID')
GOOGLE_CLIENT_SECRET = env('GOOGLE_CLIENT_SECRET')
```

**Comment by caseyduquettesc** [comment]
9.1.x using the sentry-onbuild docker image -
```python
# Google Single Sign On
SENTRY_OPTIONS['auth-google.client-id']  = env('GOOGLE_CLIENT_ID')
SENTRY_OPTIONS['auth-google.client-secret'] = env('GOOGLE_CLIENT_SECRET')
```
It's DevJackSmith's solution, but passed in with environment variables.

**Comment by caseyduquettesc** [comment]
Oddly, if I install both `sentry-auth-google==0.2.0` and `sentry-plugins==9.1.1`, then the `SENTRY_OPTIONS` solution no longer works, resulting in a "Missing client id" error on the Google sign in page.

After giving up trying to deduce the conflict, I started doing some trial and error. Oddly, the original instructions in the readme are the only ones that worked, meaning set `GOOGLE_CLIENT_ID`. 

I am not sure why installing `sentry-plugins` voided my configuration (or maybe some would call this making it work?), but this is what I ended up using to cover all my bases with or without sentry-plugins:

```python
# Google Single Sign On
GOOGLE_OAUTH2_CLIENT_ID = env('GOOGLE_CLIENT_ID')
GOOGLE_OAUTH2_CLIENT_SECRET = env('GOOGLE_CLIENT_SECRET')
GOOGLE_CLIENT_ID = GOOGLE_OAUTH2_CLIENT_ID
GOOGLE_CLIENT_SECRET = GOOGLE_OAUTH2_CLIENT_SECRET
SENTRY_OPTIONS['auth-google.client-id'] = GOOGLE_OAUTH2_CLIENT_ID
SENTRY_OPTIONS['auth-google.client-secret'] = GOOGLE_OAUTH2_CLIENT_SECRET
```

**Comment by DevJackSmith** [comment]
i think sentry plugins now includes google auth. So no need to install them separately. (You probably don't need 2 middle lines. The two top lines are just passing it from the environment variables into local variables and then again into local SENTRY_OPTIONS variables, so you might simplify it with just the two lines like this https://github.com/getsentry/sentry-auth-google#issuecomment-490260014 and it should still function just fine. If readme instructions work, i'm all for closing this issue. Can someone else confirm the readme instructions work for them now?

**Comment by caseyduquettesc** [comment]
Does the plugin package include google auth? I didn't see it here https://github.com/getsentry/sentry-plugins/tree/master/src/sentry_plugins

If it's in there though, I have no problem uninstalling this package. Last I heard though, they hadn't merged the auth plugins with the rest of the plugins. I don't recall which thread I saw it in, but it sounded like the team hadn't gotten around to merging them yet.

Though what you're saying is the only answer that makes sense why installing sentry-plugins changes the behavior of sentry-auth-google.

**Comment by DevJackSmith** [comment]
You are right, i don't know if google-auth is included in sentry-plugins now, i just assumed it was since I had to remove the zip url from requirements.txt file, and I created the server from scratch without it. So it must be in there somewhere unless I missed something.

**Comment by gordcorp** [comment]
FYI I happened to see this plugin **is** now included in sentry core now @DevJackSmith @caseyduquettesc  https://github.com/getsentry/sentry/pull/11769

**Comment by okainov** [comment]
I gave up... This masterpiece does not work at all. Tool gitlab SSO https://github.com/SkyLothar/sentry-auth-gitlab which had zero update since two years and it worked out-of-the box instantly.
